types/node/globals.d.ts

```diff
@@ -1132,7 +1132,12 @@ declare namespace NodeJS {
     class Module {
         static runMain(): void;
         static wrap(code: string): string;
-        static createRequireFromPath(path: string): (path: string) => any;
+
+        /**
+         * @deprecated Deprecated since: v12.2.0. Please use createRequire() instead.
+         */
+        static createRequireFromPath(path: string): NodeRequireFunction;
+        static createRequire(path: string): NodeRequireFunction;
```

Comment:
`createRequire` allows `string | URL`

Reply:
updated

Reply:
removed the url parameter. Actually I think we should import url from "url.d.ts" but I am not sure how it must be done in right way.

Reply:
iirc referencing modules in a global context doesn't work.

Reply:
`string | import('url').URL`?

Reply:
That was added in TypeScript 2.9, though. To support your TS 2.1 definition, I believe it would work to `import { URL } from "url";` at the top and then wrap the rest of the file in `declare global { ... }`.

Reply:
I tried the `declare global {}` variant but failed. it complains in process.d.ts that node typings are missing.

Off topic but I have to comment regarding "To support **_your_** TS 2.1 definition": we asked several times to raise the min typescript version for definitions here but this was rejected by the maintainers of this repo.
Doing it by hand means updating >1000 dependents and this ends up in a PR that unicorns because of a too big changeset and as a result gets never merged as it can't be reviewed.

Reply:
Ugh, it looks like the thing that’s preventing `declare global` is a bug that ran all the way up until 2.9: https://github.com/microsoft/TypeScript/issues/23796